File: submissions/yapzukai/python/legacy_code.py
"""
Legacy Python 2.7 Code (BEFORE Modernization)
==============================================

This file shows the original legacy code that was modernized.
Included for comparison and educational purposes.

PROBLEMS WITH THIS CODE:
- Python 2.7 syntax (print statements, string formatting)
- urllib2 (deprecated HTTP library)
- No error handling
- No async support
- Manual JSON handling
- No type hints
- No logging
- No configuration management
- No authentication
- No retry logic
- Security vulnerabilities
"""

import urllib.request
import urllib.error
import json

def transfer_money(from_acc, to_acc, amount):
    # Old-style string formatting
    url = "http://localhost:8123/transfer"
    
    # Manual JSON encoding
    data = '{"fromAccount":"' + from_acc + '","toAccount":"' + to_acc + '","amount":' + str(amount) + '}'
    
    # Old urllib approach (updated for Python 3)
    req = urllib.request.Request(url, data.encode('utf-8'))
    req.add_header('Content-Type', 'application/json')
    
    try:
        response = urllib.request.urlopen(req)
        result = response.read().decode('utf-8')
        print("Transfer result: " + result)
        return result
    except urllib.error.HTTPError as e:
        print("Error: " + str(e.code))
        return None

# Usage - old style
if __name__ == "__main__":
    transfer_money("ACC1000", "ACC1001", 100.00)

File: submissions/yapzukai/javascript/legacy-code.js
/**
 * Legacy ES5 JavaScript Code (BEFORE Modernization)
 * =================================================
 * 
 * This file shows the original legacy code that was modernized.
 * Included for comparison and educational purposes.
 * 
 * PROBLEMS WITH THIS CODE:
 * - ES5 syntax (var, function declarations)
 * - XMLHttpRequest (old HTTP API)
 * - Synchronous requests (blocking)
 * - Manual JSON string building
 * - No error handling
 * - No authentication
 * - No retry logic
 * - No input validation
 * - No logging
 * - Security vulnerabilities
 * - Poor user experience
 */

// Legacy ES5 JavaScript - MODERNIZE THIS!
function transferMoney(fromAccount, toAccount, amount) {
    // Old XMLHttpRequest approach
    var xhr = new XMLHttpRequest();
    var url = "http://localhost:8123/transfer";
    
    // Manual JSON string building - DANGEROUS!
    var data = '{"fromAccount":"' + fromAccount + '","toAccount":"' + toAccount + '","amount":' + amount + '}';
    
    xhr.open("POST", url, false); // Synchronous - BAD!
    xhr.setRequestHeader("Content-Type", "application/json");
    
    try {
        xhr.send(data);
        if (xhr.status == 200) {
            var result = JSON.parse(xhr.responseText);
            console.log("Transfer successful: " + result.transactionId);
            return result;
        } else {
            console.log("Error: " + xhr.status);
            return null;
        }
    } catch (e) {
        console.log("Request failed: " + e.message);
        return null;
    }
}

// Usage - old style
transferMoney("ACC1000", "ACC1001", 100.00);